tests/common/dnspolicy/dnspolicy_controller_test.go

```diff
@@ -768,41 +802,199 @@ var _ = Describe("DNSPolicy controller", func() {
 			}, tests.TimeoutMedium, tests.RetryIntervalMedium).Should(Succeed())
 
-			err := k8sClient.Delete(ctx, gateway)
-			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+			By("changing the policy target ref")
+			Eventually(func() error {
+				existingDNSpolicy := &v1alpha1.DNSPolicy{}
+				if err := k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsPolicy), existingDNSpolicy); err != nil {
+					return err
+				}
+				patch := client.MergeFrom(existingDNSpolicy.DeepCopy())
+				existingDNSpolicy.Spec.TargetRef.Name = "doesnotexist"
+				return k8sClient.Patch(ctx, existingDNSpolicy, patch)
+			}, tests.TimeoutMedium, time.Second).Should(Succeed())
 
-			err = k8sClient.Delete(ctx, dnsPolicy)
-			Expect(client.IgnoreNotFound(err)).ToNot(HaveOccurred())
+			Eventually(func(g Gomega) { // DNS records(s) do not exist and gateway back references removed
+				g.Expect(k8sClient.Get(ctx, client.ObjectKey{Name: recordName, Namespace: testNamespace}, &kuadrantdnsv1alpha1.DNSRecord{})).Should(MatchError(ContainSubstring("not found")))
+				g.Expect(k8sClient.Get(ctx, client.ObjectKey{Name: wildcardRecordName, Namespace: testNamespace}, &kuadrantdnsv1alpha1.DNSRecord{})).Should(MatchError(ContainSubstring("not found")))
 
-			//ToDo We cant assume that a dnsrecord reconciler is running that will remove the record or that it will be removed
-			// It's not the responsibility of this operator to do this, so we should just check if it's gone
-			// (in case we are running on a cluster that actually has a dnsrecord reconciler running), or that it is marked for deletion
-			//Eventually(func() error { // DNS record removed
-			//	return k8sClient.Get(ctx, client.ObjectKey{Name: recordName, Namespace: testNamespace}, &kuadrantdnsv1alpha1.DNSRecord{})
-			//}, tests.TimeoutMedium, tests.RetryIntervalMedium).Should(MatchError(ContainSubstring("not found")))
+				//ToDo (mnairn) Theres a bug here, the old gateway target never has these annotations removed
+				//err := k8sClient.Get(ctx, client.ObjectKeyFromObject(gateway), gateway)
+				//g.Expect(err).NotTo(HaveOccurred())
+				//g.Expect(gateway.Annotations).ToNot(HaveKey(v1alpha1.DNSPolicyDirectReferenceAnnotationName))
+				//g.Expect(gateway.Annotations).ToNot(HaveKeyWithValue(v1alpha1.DNSPolicyBackReferenceAnnotationName, policiesBackRefValue))
+
+				//Check policy status
+				err := k8sClient.Get(ctx, client.ObjectKeyFromObject(dnsPolicy), dnsPolicy)
+				g.Expect(err).NotTo(HaveOccurred())
+				g.Expect(dnsPolicy.Status.Conditions).To(
+					ConsistOf(
+						MatchFields(IgnoreExtras, Fields{
+							"Type":   Equal(string(gatewayapiv1alpha2.PolicyConditionAccepted)),
+							"Status": Equal(metav1.ConditionFalse),
+							"Reason": Equal("TargetNotFound"),
+						}),
+					),
+				)
+				//ToDo (mnairn) Theres a bug here, this never gets reset when all records are removed due to target removal
+				//g.Expect(dnsPolicy.Status.TotalRecords).To(Equal(int32(0)))
```

Comment:
Issues with the record count will be resolved by the SOTW status reconciler updates.